cmds/vboot/vboot.go

```diff
@@ -21,52 +20,45 @@ const (
 	filesystem string = "ext3"
 )
 
-func dieHard() {
-	if e1, e2, err := syscall.Syscall6(syscall.SYS_REBOOT, syscall.LINUX_REBOOT_MAGIC1, syscall.LINUX_REBOOT_MAGIC2, syscall.LINUX_REBOOT_CMD_POWER_OFF, 0, 0, 0); err != 0 {
-		log.Fatalf("a %v b %v err %v", e1, e2, err)
+var (
+	publicKey            = flag.String("pubkey", "/etc/sig.pub", "A public key which should verify the signature.")
+	pcr                  = flag.Uint("pcr", 12, "The pcr index used for measuring the kernel before kexec.")
+	bootDev              = flag.String("boot-device", "/dev/sda1", "The boot device which is used to kexec into a signed kernel.")
+	linuxKernel          = flag.String("kernel", "/mnt/vboot/kernel", "Kernel image file path.")
+	linuxKernelSignature = flag.String("kernel-sig", "/mnt/vboot/kernel.sig", "Kernel image signature file path.")
+	initrd               = flag.String("initrd", "/mnt/vboot/initrd", "Initrd file path.")
+	initrdSignature      = flag.String("initrd-sig", "/mnt/vboot/initrd.sig", "Initrd signature file path.")
+	debug                = flag.Bool("debug", false, "Enables debug mode.")
+	noTPM                = flag.Bool("no-tpm", false, "Disables tpm measuring process.")
+)
+
+func die(err error) {
+	if *debug {
+		panic(err)
+	}
+	if err := syscall.Reboot(syscall.LINUX_REBOOT_CMD_POWER_OFF); err != nil {
+		log.Fatalf("reboot err: %v", err)
 	}
 }
 
 func main() {
-	var publicKey = flag.String("pubkey", "/etc/sig.pub", "A public key which should verify the signature.")
-	var pcr = flag.Uint("pcr", 12, "The pcr index used for measuring the kernel before kexec.")
-	var bootDev = flag.String("boot-device", "/dev/sda1", "The boot device which is used to kexec into a signed kernel.")
-	var linuxKernel = flag.String("kernel", "/mnt/vboot/kernel", "Kernel image file path.")
-	var linuxKernelSignature = flag.String("kernel-sig", "/mnt/vboot/kernel.sig", "Kernel image signature file path.")
-	var initrd = flag.String("initrd", "/mnt/vboot/initrd", "Initrd file path.")
-	var initrdSignature = flag.String("initrd-sig", "/mnt/vboot/initrd.sig", "Initrd signature file path.")
-	var debug = flag.Bool("debug", false, "Enables debug mode.")
-	var noTPM = flag.Bool("no-tpm", false, "Disables tpm measuring process.")
-
 	flag.Parse()
 
 	if err := os.MkdirAll(mountPath, os.ModePerm); err != nil {
-		if *debug {
-			panic(err)
-		} else {
-			dieHard()
-		}
+		die(err)
 	}
 
-	if err := unix.Mount(*bootDev, mountPath, filesystem, unix.MS_RDONLY, ""); err != nil {
-		if *debug {
-			panic(err)
-		} else {
-			dieHard()
-		}
+	if err := syscall.Mount(*bootDev, mountPath, filesystem, syscall.MS_RDONLY, ""); err != nil {
+		die(err)
 	}
 
-	paths := [...]string{*publicKey, *linuxKernel, *linuxKernelSignature, *initrd, *initrdSignature}
+	paths := []string{*publicKey, *linuxKernel, *linuxKernelSignature, *initrd, *initrdSignature}
```

Comment:
why is this now a slice instead of an array? is it getting added to somewhere?

Reply:
I just saw no reason for it to be an array. I think it's fine either way? Slices are just more common.

Reply:
The reason I ask is that in some of my contributions to go upstream they have explicitly requested that I use the [...] form instead of a slice where it would work. So were this going to upstream Go you'd probably get that request.